cli/src/pcluster/templates/cluster_stack.py

```diff
@@ -607,6 +608,20 @@ def _add_shared_storage(self, storage):
     def _add_fsx_storage(self, id: str, shared_fsx: SharedFsx):
         """Add specific Cfn Resources to map the FSX storage."""
         fsx_id = shared_fsx.file_system_id
+        self.shared_storage_attributes[shared_fsx.shared_storage_type]["MountName"] = ",".join(
```

Comment:
`_add_fsx_storage` is a function to add a single fsx storage. We shouldn't loop for all the Fsx in `self.config.shared_storage`.

If you want to keep the "multi-fsx" behavior you can move these loops at the end, after finishing `_add_shared_storage` loop, but multiple FSx is not supported yet so you can simply do:
```
self.shared_storage_options[shared_fsx.shared_storage_type]["MountName"] = fs.existing_mount_name
self.shared_storage_options[shared_fsx.shared_storage_type]["DNSName"] = fs.existing_dns_name
```